### PR TITLE
PER-42 - Enhancement add meaningful hostname for AWS Instances

### PR DESCRIPTION
**Pull request by dougortiz**

### CODE REVIEW

File: edbdeploy/data/terraform/aws/environments/ec2/add_nickname_to_shell.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+if [ "$#" -lt 3 ]; then
+   echo "You must enter at least 23command line arguments"
+else
+   ARGSCOUNTER=0
+   COUNTER=0
+   for ip in "$@"
+   do
+      if [ "$ARGSCOUNTER" -gt 1 ]; then
+         echo "$ip" "$NODETYPE""$COUNTER"."$CLUSTERNAME".internal | sudo tee -a /etc/hosts
```
Comment (jt-edb) [inline edbdeploy/data/terraform/aws/environments/ec2/add_nickname_to_shell.sh:14]: Do we really want to update /etc/hosts once again? `add_ips_to_hosts.sh` is already in charge of doing this.

File: edbdeploy/data/terraform/aws/environments/ec2/add_nickname_to_shell.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+         # Configure shell prompt
+         echo "export NICKNAME=""$NODETYPE""$COUNTER"."$CLUSTERNAME".internal | sudo tee -a /etc/profile.d/prompt.sh
+         sudo sed -i "s/\\h \\W/$NODETYPE$COUNTER \\W/g" /etc/bashrc
+         if [ -f /etc/bash.bashrc ]; then
+            sudo sed -i "s/\\h \\W/$NODETYPE$COUNTER \\W/g" /etc/bash.bashrc
+         fi
+         # Remove extra characters in shell prompt
+         sudo sed -i "s/ WW//" /etc/bashrc
+         if [ -f /etc/bash.bashrc ]; then
+            sudo sed -i "s/ WW//" /etc/bash.bashrc
```
Comment (jt-edb) [inline edbdeploy/data/terraform/aws/environments/ec2/add_nickname_to_shell.sh:24]: Not sure where it comes, but I'm getting an unexpected `\` char:

```console
[centos@\pooler0]$
```

**Review by jt-edb — COMMENTED** [review]
Hi Doug,

Just tried with EDB-RA-3:

- `provision` step failed with the following error, right after `postgres_setup_volume`  execution:
```
Error: error executing "/tmp/terraform_1870469383.sh": Process exited with status 127
```
- prompt on **all** postgres nodes is the same: `[centos@\pg0]$` but we should get `[centos@pg0]$`, `[centos@pg1]$` and `[centos@pg2]$`
- same issue for pooler nodes

File: edbdeploy/data/terraform/aws/environments/ec2/add_nickname_to_shell.sh (hunk 1 above)
Comment (dougortiz) [reply edbdeploy/data/terraform/aws/environments/ec2/add_nickname_to_shell.sh:14]: Adding it via the 'add_ips_to_hosts.sh' causes duplication, so it needs to be accomplished individually

File: edbdeploy/data/terraform/aws/environments/ec2/add_nickname_to_shell.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+CLUSTERNAME=$1
+NODETYPE=$2
+
+if [ "$#" -lt 3 ]; then
+   echo "You must enter at least 23command line arguments"
```
[the remark this answers is not in the document]
Comment (dougortiz) [reply edbdeploy/data/terraform/aws/environments/ec2/add_nickname_to_shell.sh:7]: This is a typo

File: edbdeploy/data/terraform/aws/environments/ec2/ec2.tf [hunk 4]
```diff
@@ -448,7 +530,9 @@ resource "null_resource" "pg-set-hostname" {
         "/tmp/add_ips_to_hosts.sh ${var.cluster_name} pem ${join(" ", aws_instance.pem_server.*.private_ip)} >> /tmp/pem_hostnames.log 2>&1",
         "/tmp/add_ips_to_hosts.sh ${var.cluster_name} postgres ${join(" ", aws_instance.postgres_server.*.private_ip)} >> /tmp/postgres_hostnames.log 2>&1",        
         "/tmp/add_ips_to_hosts.sh ${var.cluster_name} barman ${join(" ", aws_instance.barman_server.*.private_ip)} >> /tmp/barman_hostnames.log 2>&1",
-        "/tmp/add_ips_to_hosts.sh ${var.cluster_name} pooler ${join(" ", aws_instance.pooler_server.*.private_ip)} >> /tmp/pooler_hostnames.log 2>&1"        
+        "/tmp/add_ips_to_hosts.sh ${var.cluster_name} pooler ${join(" ", aws_instance.pooler_server.*.private_ip)} >> /tmp/pooler_hostnames.log 2>&1",
+        "chmod a+x /tmp/add_nickname_to_shell.sh",        
+        "/tmp/add_nickname_to_shell.sh ${var.cluster_name} postgres ${join(" ", aws_instance.postgres_server.*.private_ip)} >> /tmp/pem_nicknames.log 2>&1"
```
Comment (jt-edb) [inline edbdeploy/data/terraform/aws/environments/ec2/ec2.tf:535]: Log file name should be `postgres_nicknames.log`

**Review by jt-edb — COMMENTED** [review]
LGTM! Only one minor comment. Thank you Doug.

File: edbdeploy/data/terraform/aws/environments/ec2/ec2.tf (hunk 4 above)
Comment (dougortiz) [reply edbdeploy/data/terraform/aws/environments/ec2/ec2.tf]: Thank you for reviewing.

FIxed in PER-42v3, ready for review and merge.

**Review by jt-edb — APPROVED** [review]
LGTM!

**Pull request merged**